### PR TITLE
feat(HII-3812): add token cache demo app

### PR DESCRIPTION
**Pull request by mdovhopo**

### CODE REVIEW

File: token-cache/README.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# Token Cache demo app
+
+This project was bootstrapped with [@hiiretail/cra-template](https://github.com/extenda/hiiretail-react-template)
+
+TODO: add description
+
+## Start the demo
+
+TODO: start steps.. (yarn start, ocms token etc)
```
Comment (l0kal) [inline token-cache/README.md:9]: Readme is missing some headers :smiling_face_with_tear:

File: token-cache/cloud-run.yaml [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+platform:
+  gke:
+    connectivity: external
+    domain-mappings:
+      prod:
+        - app-name.retailsvc.com
+      staging:
+        - app-name.retailsvc.com
+    opa-enabled: false
```
Comment (l0kal) [inline token-cache/cloud-run.yaml:14]: Will this be deployed? If it will please set the min-instance to 0 and fix the name and URLs

File: token-cache/src/common/utils/getPermissions.tsx [hunk 3]
```diff
@@ -0,0 +1,8 @@
+export const getPermissions = async (): Promise<string[]> => {
+  // const token = await firebase.app().auth().currentUser?.getIdToken();
+  // const tenantId = localStorage.getItem('tenantId') as string;
```
Comment (l0kal) [inline token-cache/src/common/utils/getPermissions.tsx:3]: remove comments?

File: token-cache/src/react-app-env.d.ts [hunk 4]
```diff
@@ -0,0 +1 @@
+/// <reference types="react-scripts" />
```
Comment (l0kal) [inline token-cache/src/react-app-env.d.ts:1]: is this file needed?

Comment (mdovhopo) [reply token-cache/src/react-app-env.d.ts:1]: Generated by template, I'll leave it

File: token-cache/src/common/utils/getPermissions.tsx (hunk 3 above)
Comment (mdovhopo) [reply token-cache/src/common/utils/getPermissions.tsx:3]: Sure :)

File: token-cache/cloud-run.yaml (hunk 2 above)
Comment (mdovhopo) [reply token-cache/cloud-run.yaml:14]: No, will remove it

File: .vscode/settings.json [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+    "[javascript]": {
+        "editor.defaultFormatter": "esbenp.prettier-vscode"
+      },
+    "[typescript]": {
+        "editor.defaultFormatter": "esbenp.prettier-vscode"
+    },
+    "[typescriptreact]": {
+        "editor.defaultFormatter": "esbenp.prettier-vscode"
+    }
+}
```
Comment (glebbash) [inline .vscode/settings.json:11]: Ignore this, file also it is better to format with eslint that uses prettier.

**Review by glebbash — APPROVED** [review]

File: .vscode/settings.json (hunk 5 above)
Comment (mdovhopo) [reply .vscode/settings.json:11]: hm ok)

Comment (mdovhopo) [reply .vscode/settings.json:11]: but for the frontend apps that does not work, so i have to use prettier)

**Review by l0kal — APPROVED** [review]
